luarules/gadgets/ai/shard_runtime/spring_lua/unit.lua

```diff
@@ -375,8 +375,11 @@ function ShardUnit:MaxWeaponsRange()
 end
 
 
-function ShardUnit:CanBuild( type )
-	return self:Type():CanBuild(type)
+function ShardUnit:CanBuild( uType )
+	if type(uType ) == 'string' then
+		uType = game:GetTypeByName(uType)
+	end
+	return self:Type():CanBuild(uType)
```

Comment:
this change shouldn't be needed, `CanBuild` onn the unit type should already do this check